### PR TITLE
[SDCOSMO-1956] Fix Runner child doesn't inherit parent's parameters

### PR DESCRIPTION
**Pull request by jreynard-code**

### CODE REVIEW

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt [hunk 1]
```diff
@@ -256,6 +287,57 @@ class RunnerService(
...
+          solution
+              ?.parameterGroups
+              ?.filter { parameterGroup ->
+                runTemplate?.parameterGroups?.contains(parameterGroup.id) == true
+              }
+              ?.flatMap { parameterGroup -> parameterGroup.parameters ?: mutableListOf() }
+      if (!runTemplateParametersIds.isNullOrEmpty()) {
+        val parentParameters = parent.parametersValues?.associate { it.parameterId to it }
+        val runnerParameters = runner.parametersValues?.associate { it.parameterId to it }
+        // TODO Handle default value
```
Comment (sjoubert) [inline runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt:312]: Comment to be removed as discussed IRL

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt [hunk 2]
```diff
@@ -256,6 +287,57 @@ class RunnerService(
...
+                    "Parameter $parameterId not found in parent ($parentId) parameters values")
+              }
+            } else {
+              logger.debug(
+                  "Skipping parameter $parameterId, defined neither in the parent nor in this Runner")
+            }
+          } else {
+            logger.debug(
+                "Skipping parameter $parameterId since it is already defined in this Runner")
+          }
```
Comment (sjoubert) [inline runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt:335]: I'm not a big fan on nested if-else for error handling. I find sequential if-return easier to read and understand.

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt (hunk 1 above)
Comment (jreynard-code) [reply runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt]: As last minute conclusion, TODO was modified to contain "Handle default value" context.

**Review by sjoubert — APPROVED** [review]

File: runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt (hunk 2 above)
Comment (jreynard-code) [reply runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt]: Rewrite using filters lambda ;)

**Pull request merged**